### ISSUE TITLE
[studio] Studio crashes when field slug is missing source option

### ISSUE THREAD

**Issue by mrkolby** [sanity-io/sanity#1652]
**Describe the bug**

The Studio crashes when `slug` field is missing a `source` option after upgrading to 0.147.0.

**To Reproduce**

Use the `slug` field without a `source` option.

#### Not working

```javascript
{
  name: 'slug',
  title: 'Slug',
  type: 'slug',
  options: {
    maxLength: 100
  }
},
```

#### Working

```javascript
{
  name: 'slug',
  title: 'Slug',
  type: 'slug',
  options: {
    maxLength: 100,
    source: 'title',
  }
},
```

**Expected behavior**

Not crashing.

**Screenshots**

<img width="1361" alt="Screenshot 2019-12-09 at 22 13 43" src="https://user-images.githubusercontent.com/300595/70473366-2b969f80-1ad1-11ea-8614-3f4e89bc7a7b.png">

**Which versions of Sanity are you using?**

```
@sanity/base               0.147.0 (up to date)
@sanity/components         0.147.0 (up to date)
@sanity/core               0.147.0 (up to date)
@sanity/default-layout     0.147.0 (up to date)
@sanity/default-login      0.147.0 (up to date)
@sanity/desk-tool          0.147.0 (up to date)
@sanity/google-maps-input  0.147.0 (up to date)
@sanity/vision             0.147.0 (up to date)
```

**What operating system are you using?**

MacOS Catalania 10.15.1

**Which versions of Node.js / npm are you running?**

```
6.13.2
v13.2.0
```

**Additional context**

Tested with a clean `sanity init` with the optional Movie project (schema). I feel like pointing to #1646 without further testing.

**Comment by rexxars** [comment]
Thanks for reporting! Looking into this now.

**Comment by rexxars** [comment]
I just released v0.147.1 which should fix this issue - thanks again for reporting! 
Run  `sanity upgrade` in your studio folder to upgrade.
